tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp

```diff
@@ -360,3 +387,269 @@ TEST_F(CRTLoggingTest, testTraceLogLevel)
 {
     DoCRTLogTest(LogLevel::Trace);
 }
+
+class LoggingTestLogFileRace : public Aws::Testing::AwsCppSdkGTestSuite
+{
+};
+
+void LogOnCRTLogSystemInterfaceWithoutVaArgs(CRTLogSystemInterface* object, LogLevel logLevel, const char* subjectName, const char* formatStr, ...)
+{
+    va_list args;
+    va_start(args, formatStr);
+    assert(object);
+    object->Log(logLevel, subjectName, formatStr, args);
+    va_end(args);
+}
+
+TEST_F(LoggingTestLogFileRace, testRaceOnLogFile)
+{
+    /*
+     * Tests a race condition on the default log file (in case of multiple processes / loggers using the same SDK log file).
+     */
+    static const size_t PARALLEL_COUNT = 32;
```

Comment:
perhaps use `floor (std::thread::hardware_concurrency, 32)` for when we run on restricted envs